.github/workflows/unit_test_matrix.yml

```diff
@@ -0,0 +1,76 @@
+name: Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  unit_tests:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        python_version: ${{ matrix.python_versions }}
+        tf_version: ${{ matrix.tf_versions }}
+        onnx_opset: ${{ matrix.onnx_opsets }}
+        include:
+          - python_versions: ['3.8']
+            tf_versions: ['2.9.0']
+            onnx_opsets: ['17', '16', '14']
+            platforms: ['linux', 'windows']
```

Comment:
I think you need `ubuntu-latest` and `windows-latest`: https://github.com/onnx/onnx/blob/093a8d335a66ea136eb1f16b3a1ce6237ee353ab/.github/workflows/main.yml#L30

Reply:
Windows-latest is not always stable for TF tests.